src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -164,9 +115,29 @@ public static object CreateEnumerableValue(ref Utf8JsonReader reader, ref ReadSt
             }
         }
 
+        public Type GetElementType()
+        {
+            if (IsPropertyEnumerable)
+            {
+                return JsonPropertyInfo.ElementClassInfo.Type;
+            }
+
+            if (IsEnumerable)
+            {
+                return JsonClassInfo.ElementClassInfo.Type;
+            }
+
+            if (IsDictionary)
+            {
+                return JsonClassInfo.ElementClassInfo.Type;
+            }
+
+            return JsonPropertyInfo.RuntimePropertyType;
```

Comment:
Having multiple return statements like this within a method (especially one called in hot loops) could be slightly problematic for perf. Consider storing the result in a local with a single return at the end.